spec/workers/articles/publish_worker_spec.rb

```diff
@@ -1,104 +1,120 @@
 require "rails_helper"
 
 RSpec.describe Articles::PublishWorker, type: :worker do
+  let(:worker) { subject }
+  let!(:article) { create(:article, published: true) }
+
   include_examples "#enqueues_on_correct_queue", "medium_priority"
 
-  describe "#perform" do
-    let(:worker) { subject }
-    let!(:article) { create(:article, published: true) }
+  it "calls Slack::Messengers::ArticlePublished to send slack notifications" do
+    allow(Slack::Messengers::ArticlePublished).to receive(:call)
+    worker.perform
+    expect(Slack::Messengers::ArticlePublished).to have_received(:call).with(article: article)
+  end
 
-    it "calls Slack::Messengers::ArticlePublished to send slack notifications" do
-      allow(Slack::Messengers::ArticlePublished).to receive(:call)
-      worker.perform
-      expect(Slack::Messengers::ArticlePublished).to have_received(:call).with(article: article)
-    end
+  it "sends notifications to mentioned users and followers" do
+    allow(Notification).to receive(:send_to_mentioned_users_and_followers)
+    worker.perform
+    expect(Notification).to have_received(:send_to_mentioned_users_and_followers).with(article)
+  end
 
-    it "sends notifications to mentioned users and followers" do
-      allow(Notification).to receive(:send_to_mentioned_users_and_followers)
+  it "schedules Notifications::NotifiableActionWorker" do
+    args = [article.id, "Article", "Published"]
+    sidekiq_assert_enqueued_with(job: Notifications::NotifiableActionWorker, args: args) do
       worker.perform
-      expect(Notification).to have_received(:send_to_mentioned_users_and_followers).with(article)
     end
+  end
+
+  it "doesn't send notifications for an old article" do
+    old_article = create(:article, :past, published: true, past_published_at: 1.year.ago)
+    allow(Notification).to receive(:send_to_mentioned_users_and_followers)
+    worker.perform
+    expect(Notification).not_to have_received(:send_to_mentioned_users_and_followers).with(old_article)
+  end
+
+  it "doesn't send notifications for a scheduled article" do
+    scheduled_article = create(:article, published: true, published_at: 1.day.from_now)
+    allow(Notification).to receive(:send_to_mentioned_users_and_followers)
+    worker.perform
+    expect(Notification).not_to have_received(:send_to_mentioned_users_and_followers).with(scheduled_article)
+  end
 
-    it "schedules Notifications::NotifiableActionWorker" do
-      args = [article.id, "Article", "Published"]
-      sidekiq_assert_enqueued_with(job: Notifications::NotifiableActionWorker, args: args) do
+  it "sends moderation notification for a published article" do
+    unpublished = create(:unpublished_article, user: article.user)
+    allow(Notification).to receive(:send_moderation_notification)
+    worker.perform
+    expect(Notification).to have_received(:send_moderation_notification).with(article)
+    expect(Notification).not_to have_received(:send_moderation_notification).with(unpublished)
```

Comment:
Here we're also testing that we don't send for an unpublished article, I think that's worth a separate spec.

Reply:
I waffled a bit on this. In general, I get nervous about false-positives when we assert `not_to have_received` (because there are lots of ways to "not send" a message that don't actually mean the code is performing as expected). 

Ideally, I would have preferred to have this test without the `let!` on 5, (because we can then assert that **no** notifications were sent — isn't perfect, but cuts the potential for false confidence roughly in half), but I was also trying to minimize the footprint here (partly because I was originally aiming at the flaky test).

Having a think about it this morning, along with the suggestion of adding a spec for `scheduled` article, I think splitting them into separate specs is better for future readers, even with some risks of false confidence. :+1: